### PR TITLE
Initial support for continuous signals

### PR DESCRIPTION
**Pull request by rphh**

### CODE REVIEW

File: sdl2promela/sdl/model.py [hunk 1]
```diff
@@ -972,12 +986,26 @@ def __gather_states(self):
...
     def __get_inputs_of_name(self, name: str) -> List[ogAST.Input]:
         result = set()
         for inputs in self.source.mapping.values():
             if isinstance(inputs, List):
                 for input in inputs:
-                    if isinstance(input, ogAST.Input) and (name in input.inputlist):
+                    if (
+                        isinstance(input, ogAST.Input)
+                        and (name in input.inputlist)
+                        and len(input.inputlist) == 1
```
Comment (Lurkerpas) [inline sdl2promela/sdl/model.py:1007]: Why == 1?

File: sdl2promela/stop_condition/translator.py [hunk 2]
```diff
@@ -490,6 +490,61 @@ def _generate_exist(context: GenerateContext, expr: model.CallExpression):
     )
 
 
+def _get_queue_name(context: GenerateContext, expr: model.CallExpression):
+    if len(expr.parameters) != 1:
+        raise TranslateException("Function 'empty' requires one parameter")
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:495]: Why empty? How is it related to the name of the method?

File: sdl2promela/stop_condition/translator.py [hunk 3]
```diff
@@ -490,6 +490,61 @@ def _generate_exist(context: GenerateContext, expr: model.CallExpression):
     )
 
 
+def _get_queue_name(context: GenerateContext, expr: model.CallExpression):
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:493]: As it is now, it is hard to tell what is the purpose of this function. Could you please rename it and provide some comments?

File: sdl2promela/stop_condition/translator.py [hunk 4]
```diff
@@ -490,6 +490,61 @@ def _generate_exist(context: GenerateContext, expr: model.CallExpression):
...
+        raise TranslateException(
+            "Cannot find queue with name '{}' in process '{}'".format(
+                queue_name, process_name
+            )
+        )
+
+    return "{}_{}_channel".format(process_name.lower(), queue_name.lower())
+
+
+def _generate_empty(context: GenerateContext, expr: model.CallExpression):
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:528]: _generate_call_to_empty? _generate_emptiness_query?

File: sdl2promela/stop_condition/translator.py [hunk 5]
```diff
@@ -490,6 +490,61 @@ def _generate_exist(context: GenerateContext, expr: model.CallExpression):
...
+    queue_variable = _get_queue_name(context, expr)
+    return (
+        promelaBuilder.CallBuilder()
+        .withTarget("empty")
+        .withParameter(promelaBuilder.VariableReferenceBuilder(queue_variable).build())
+        .build()
+    )
+
+
+def _generate_queue_length(context: GenerateContext, expr: model.CallExpression):
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:538]: _generate_queue_length_call/query?

File: sdl2promela/sdl/model.py [hunk 6]
```diff
@@ -230,6 +230,16 @@ def __init__(self):
         self.transitions = {}
 
 
+class ContinuousSignal:
+
```
Comment (Lurkerpas) [inline sdl2promela/sdl/model.py:234]: Docs

File: sdl2promela/stop_condition/translator.py (hunk 2 above)
Comment (Lurkerpas) [reply sdl2promela/stop_condition/translator.py:495]: This is also used in queue_length method....

File: sdl2promela/sdl/model.py (hunk 1 above)
Comment (rphh) [reply sdl2promela/sdl/model.py:1007]: To get proper number of signal parameters: The `*` or `StateA,StateB,StateX` cannot be used to get valid parameter list.

File: sdl2promela/stop_condition/translator.py (hunk 2 above)
Comment (rphh) [reply sdl2promela/stop_condition/translator.py:495]: Ou, this is a Stop Condition which I've implemented, but it needs proper tests...

File: sdl2promela/stop_condition/translator.py (hunk 3 above)
Comment (rphh) [reply sdl2promela/stop_condition/translator.py]: Renamed

File: sdl2promela/stop_condition/translator.py (hunk 4 above)
Comment (rphh) [reply sdl2promela/stop_condition/translator.py]: Changed.

File: sdl2promela/stop_condition/translator.py (hunk 5 above)
Comment (rphh) [reply sdl2promela/stop_condition/translator.py]: Changed.

File: sdl2promela/sdl/model.py (hunk 6 above)
Comment (rphh) [reply sdl2promela/sdl/model.py:235]: Added.

File: sdl2promela/sdl/model.py (hunk 1 above)
Comment (Lurkerpas) [reply sdl2promela/sdl/model.py:1010]: OK, so I think this needs a comment and possibly TODO

Comment (rphh) [reply sdl2promela/sdl/model.py:1014]: Added: see last commit

**Pull request merged**